creusot/src/validate/terminates.rs

```diff
@@ -450,6 +478,11 @@ impl CallGraph {
             if !(is_pearlite(ctx.tcx, def_id) || ctx.sig(def_id).contract.terminates) {
                 // Only consider functions marked with `terminates`: we already ensured
                 // that a `terminates` functions only calls other `terminates` functions.
+                if let Some(variant) = &ctx.sig(def_id).contract.variant {
+                    ctx.dcx().struct_span_warn(variant.span, "variant will be ignored")
```

Comment:
ignored, really ? It seems to me that the variant will be checked, but it is useless to prove the termination of the function.

Reply:
It is ignored, because we don't even do the analysis in this case. Thus we don't know what recursive calls the function will make, which is needed to know where to insert the variant checks.